src/generic.jl

```diff
@@ -892,6 +892,89 @@ opnorm(v::TransposeAbsVec) = norm(v.parent)
 
 norm(v::AdjOrTrans, p::Real) = norm(v.parent, p)
 
+"""
+    lpdist(x::T, y::T, p::Int) where {T <: AbstractVector}
+    lpdist(x::T, y::T, p::Real) where {T <: AbstractVector}
+
+    Computes the Lₚ distance between two vectors `x` and `y`, defined as:
+    for 0 < p < Inf: Lₚ dist = ᵖ√(Σ|xᵢ - yᵢ|ᵖ)
+    for p = 0: count of mismatches for p = 0
+    for p + Inf: L_∞ = maxᵢ(abs(xᵢ - yᵢ))
+"""
+
+function _lpunnorm(x::T, y::T, fn) where {T}
+    r = 0.0
+    for i in eachindex(x,y)
+        @inbounds r += fn(x[i] - y[i])
```

Comment:
Note that this suffers from spurious overflow if the elements of `x` or `y` are very large.

e.g. `lpdist([1e300],[0.0])` should return `1e300`, but this will return `Inf`.

Reply:
Thanks for the catch! Any pointers on how I can fix this?

Reply:
Perhaps divide each vector by the maximum absolute value before computing the distance, and rescale the result later?

Reply:
Sounds good.

Reply:
Note that rescaling is not needed for the L1 norm or the maximum norm.

Note also that you should be using `norm`, not `abs`, everywhere, in order to handle vectors whose entries are themselves vectors.  In fact, arguably you should be calling the distance function recursively for such cases.

In general, I would study the `generic_norm` implementation, which already deals with such issues.

Reply:
Okay. I intended this to be an implementation of vectors of reals (and maybe complex) only. Would it be better to specialise for real/complex vectors and have a separate implementation for recursive structures?

Reply:
See how it's done for `norm`.  There's no need to have two implementations, since `normsqr` just calls `abs2` for scalars.